src/main/java/com/google/cloud/spanner/jdbc/SpannerJdbcDialect.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (C) 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.spanner.jdbc;
+
+import static org.apache.spark.sql.types.DataTypes.BinaryType;
+import static org.apache.spark.sql.types.DataTypes.BooleanType;
+import static org.apache.spark.sql.types.DataTypes.ByteType;
+import static org.apache.spark.sql.types.DataTypes.DateType;
+import static org.apache.spark.sql.types.DataTypes.DoubleType;
+import static org.apache.spark.sql.types.DataTypes.FloatType;
+import static org.apache.spark.sql.types.DataTypes.IntegerType;
+import static org.apache.spark.sql.types.DataTypes.LongType;
+import static org.apache.spark.sql.types.DataTypes.ShortType;
+import static org.apache.spark.sql.types.DataTypes.StringType;
+import static org.apache.spark.sql.types.DataTypes.TimestampType;
+
+import org.apache.spark.sql.jdbc.JdbcDialect;
+import org.apache.spark.sql.jdbc.JdbcType;
+import org.apache.spark.sql.types.DataType;
+import org.apache.spark.sql.types.DecimalType;
+import scala.Option;
+
+public class SpannerJdbcDialect extends JdbcDialect {
+
+  private static final long serialVersionUID = 1L;
+
+  @Override
+  public boolean canHandle(String url) {
+    return url.toLowerCase().startsWith("jdbc:cloudspanner:");
+  }
+
+  @Override
+  public String quoteIdentifier(String column) {
+    return "`" + column + "`";
+  }
+
+  /** Handle spanner DDL type names, for creating tables when writing to spanner */
+  public Option<JdbcType> getJDBCType(final DataType dt) {
+    if (IntegerType.equals(dt)) {
+      return Option.apply(new JdbcType("INT64", 4));
```

Comment:
These integers are the "jdbcNullType", I kept the same integers that are used in `org.apache.spark.sql.execution.datasources.jdbc.JdbcUtils$getCommonJDBCType`. But I'm honestly not sure what they really indicate or if I could just set them all to zero or something.